src/test/scala/AWSKeySigningSpec.scala

```diff
@@ -54,18 +64,177 @@ class AWSKeySigningSpec extends FunSpec with Matchers {
+      val testSignKey = SignRequestForAWS.getSignatureKey(kSecret, "20110909", "us-east-1", "iam")
+      val testSignature = SignRequestForAWS.getSignature(testSignKey, testForm)
+      val signature = "ced6826de92d2bdeed8f846f0bf508e8559e98e4b0199114b84c54174deb456c"
+      testSignature shouldBe signature
+    }
+
+    it("Should return the correct Authorization Header"){
+      val entity = HttpEntity("Action=ListUsers&Version=2010-05-08")
+      val request = HttpRequest(
+        method = HttpMethods.POST,
+        uri = "https://iam.amazonaws.com/",
+        entity = entity,
+        headers = List(
+          RawHeader("Content-Type", "application/x-www-form-urlencoded; charset=utf-8"),
+          RawHeader("X-Amz-Date", "20110909T233600Z"))
+      )
+      val kSecret = "wJalrXUtnFEMI/K7MDENG+bPxRfiCYEXAMPLEKEY"
+      val region = "us-east-1"
+      val accessKeyID = "AKIDEXAMPLE"
+      val testAuthorization = Await.result(SignRequestForAWS.createAuthorizationHeader(request, kSecret, region, accessKeyID, "iam"), 10 seconds)
+      val  authorization = "AWS4-HMAC-SHA256 Credential=AKIDEXAMPLE/20110909/us-east-1/iam/aws4_request, SignedHeaders=content-type;host;x-amz-date, Signature=ced6826de92d2bdeed8f846f0bf508e8559e98e4b0199114b84c54174deb456c"
+      testAuthorization shouldBe authorization
+    }
+  val uuid = UUID.randomUUID().toString
+    //uses a random UUID to prevent using the same index again
+    it("Should post the correct httpRequest with all the necessary aws authentication") {
+      import DefaultJsonProtocol._
+      val accessKeyID = awsConfig.getString("accessKeyID")
+      val region = awsConfig.getString("region")
+      val kSecret = awsConfig.getString("kSecret")
+      val baseURI = awsConfig.getString("testURI")
+      val URI = s"${baseURI}test/$uuid"
+      val data = Map("item1"->"1", "item2"->"2", "item3"->"3")
+      val paramStr2 = data.toJson.compactPrint
+      val entity = HttpEntity(contentType = MediaTypes.`application/x-www-form-urlencoded`, paramStr2)
+      println(paramStr2)
+      val request = HttpRequest(
+        method = HttpMethods.POST,
+        uri = URI,
+        entity = entity
+      )
+      val authRequest = Await.result(SignRequestForAWS.addAuthorizationHeader(request, kSecret, region, accessKeyID, "es"), 10 seconds)
+      println(authRequest)
+      val response = Await.result(SignRequestForAWS.post(authRequest), 10 seconds)
+      val responseData =  Await.result(response.entity.dataBytes.map(_.utf8String).grouped(Int.MaxValue).runWith(Sink.head), 10 seconds).mkString
+      val responseJson = responseData.toJson
+      println(responseJson.prettyPrint.replace("\\\\", "\\").replace("\\n", "\n"))
+      val testForm = Await.result(SignRequestForAWS.createCanonicalRequest(request), 10 seconds)
+      println(testForm)
+      response.status.toString shouldBe "201 Created"
+    }
+
+    it ("should be a sucessful get request of the posted information") {
```

Comment:
The actual calls to AWS should be moved to an integration test, instead of a unit test to make it clear that they depend on an external service.